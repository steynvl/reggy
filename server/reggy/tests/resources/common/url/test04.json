{
  "type": "URL",
  "information": {
    "schemes": "feed;http;https;www.",
    "portNumbers": "No port number",
    "specOptionalPortNumbers": "",
    "specRequiredPortNumbers": "",
    "username": "No user names",
    "specUserNames": "",
    "password": "No password",
    "domainName": "Allow any domain name",
    "specDomainNames": "",
    "specificTld": "",
    "subdomainOnSpecDomain": "",
    "folders": "No folders",
    "minFolderDepth": "",
    "maxFolderDepth": "",
    "specFoldersOnly": "",
    "fileNames": "Allow any file name",
    "specExtensions": "",
    "specFileNames": "",
    "optionalFileNames": true,
    "parameters": "No parameters",
    "specParameters": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
