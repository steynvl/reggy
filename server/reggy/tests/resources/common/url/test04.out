Java
Pattern regex = Pattern.compile("(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?");
(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?
Perl
my $regex = /(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?/;
(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
POSIX
(?:(?:feed|https?)://|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
(?:(?:feed|https?)://|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
Python
regex = re.compile(r'(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&\'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?')
(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&\'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
JavaScript
const regex = /(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?/;
(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
PHP
$regex = '/(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?/';
(?:(?:feed|https?):\/\/|www\.)(?:[\da-z-]++\.)+[a-z]{2,63}+\/?(?:\/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
Golang
regex, _ := regexp.Compile("(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]+\\.)+[a-z]{2,63}/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?")
(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]+\\.)+[a-z]{2,63}/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?
Rust
let re = Regex::new(r"(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?").unwrap();
(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
C#
var re = new Regex(@"(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?");
(?:(?:feed|https?)://|www\.)(?:[\da-z-]+\.)+[a-z]{2,63}/?(?:/[]\d!"#$%&'()*+,.:;<=>?@\[\_`a-z{|}~^-]*)?
Scala
val re = "(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?".r
(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?
Kotlin
val regex = Regex("(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?")
(?:(?:feed|https?)://|www\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?(?:/[]\\d!"#$%&'()*+,.:;<=>?@\[\\_`a-z{|}~^-]*)?